Index: assets/js/translator.js
===================================================================
--- assets/js/translator.js	(revision 57870)
+++ assets/js/translator.js	(working copy)
@@ -153,6 +153,20 @@
             });
             $('a#fileDownload').fadeOut('fast');
         });
+
+        $('body').on('dragover', function (e) { e.preventDefault(); return false; });
+        $('body').on('dragenter', function (e) {
+            e.preventDefault();
+            if(!$('#fileDropBackdrop:visible').length) {
+                var backdrop = $('#fileDropBackdrop');
+                backdrop.fadeTo(400, 0.5);
+                backdrop.on('drop', function (e) { e.preventDefault(); console.log('read the file and  display the file uploading'); /* IMPORTANT STUFF */ });
+                backdrop.on('dragleave', function () {
+                    backdrop.fadeOut();
+                });
+            }
+            return false;
+        });
     });
 }
 
Index: index.html.in
===================================================================
--- index.html.in	(revision 57870)
+++ index.html.in	(working copy)
@@ -201,6 +201,11 @@
                                         </div>
                                   </div>
                               </div>
+                              <div class="modal-backdrop" id="fileDropBackdrop" style="display: none;">
+                                <div class="well well-lg" style="text-align: center; position: absolute; top: 35%; left: 35%;">
+                                    <h1 style="font-weight: bold;">Drop your document</h1>
+                                </div>
+                              </div>
                               <div id="translateDoc" style="display: none;">
                                   <div class="col-sm-6">
                                       <div class="well well-lg">
@@ -423,7 +428,7 @@
                                     <img alt="Creative Commons licence" data-src="./img/cc-by-sa-3.0-88x31.png">
                                 </a>
                             </div>
-                            
+
                             <div style="display: inline-block; margin: 0 .5em 0 .5em">
                                 <a href="https://www.gnu.org/licenses/gpl.html">
                                     <img alt="GNU GPL License" data-src="./img/gplv3-88x31.png">
