Index: assets/css/bootstrap.css
===================================================================
--- assets/css/bootstrap.css	(revision 52041)
+++ assets/css/bootstrap.css	(working copy)
@@ -6445,16 +6445,24 @@
   margin-left: 0;
 }
 
-@media screen and (min-width: 768px) {
+@media (min-width: 768px) {
   .modal-dialog {
     width: 600px;
     margin: 30px auto;
   }
   .modal-content {
-    -webkit-box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
-            box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
+    -webkit-box-shadow: 0 5px 15px rgba(0, 0, 0, .5);
+            box-shadow: 0 5px 15px rgba(0, 0, 0, .5);
   }
+  .modal-sm {
+    width: 300px;
+  }
 }
+@media (min-width: 992px) {
+  .modal-lg {
+    width: 900px;
+  }
+}
 
 .tooltip {
   position: absolute;
Index: assets/js/util.js
===================================================================
--- assets/js/util.js	(revision 52041)
+++ assets/js/util.js	(working copy)
@@ -28,8 +28,8 @@
             });
     }
     else {
-        $('.navbar-toggle').hide();
-        $('.nav').hide();
+        $('.navbar-default .navbar-toggle').hide();
+        $('.navbar-default .nav').hide();
     }
 
     var hash = parent.location.hash;
@@ -40,9 +40,9 @@
     }
 
     $('.modeContainer' + hash + 'Container').show();
-    $('.nav li > a[data-mode=' + hash.substring(1) + ']').parent().addClass('active');
+    $('.navbar-default .nav li > a[data-mode=' + hash.substring(1) + ']').parent().addClass('active');
 
-    $('.nav a').click(function () {
+    $('.navbar-default .nav a').click(function () {
         var mode = $(this).data('mode');
         $('.nav li').removeClass('active');
         $(this).parent('li').addClass('active');
@@ -67,10 +67,15 @@
     });
 
     $('.modal').on('show.bs.modal', function () {
+        $('a[data-target=#' + $(this).attr('id') + ']').parents('li').addClass('active');
         $.each($(this).find('img[data-src]'), function () {
             $(this).attr('src', $(this).attr('data-src'));
         });
     });
+
+    $('.modal').on('hide.bs.modal', function () {
+        $('a[data-target=#' + $(this).attr('id') + ']').parents('li').removeClass('active');
+    });
 });
 
 if(config.GOOGLE_ANALYTICS_PROPERTY && config.GOOGLE_ANALYTICS_TRACKING_ID) {
Index: index.html.in
===================================================================
--- index.html.in	(revision 52041)
+++ index.html.in	(working copy)
@@ -226,6 +226,12 @@
 
             <div class="navbar-fixed-bottom hidden-xs">
                 <div class="container">
+                    <ul class="nav nav-pills pull-left" style="display: inline-block;">
+                        <li><a data-toggle="modal" data-target="#aboutModal" data-keyboard="true"><i class="fa fa-question-circle"></i> <span data-text="About">About</span></a></li>
+                        <li><a data-toggle="modal" data-target="#downloadModal" data-keyboard="true"><i class="fa fa-download"></i> <span data-text="Download">Download</span></a></li>
+                        <li><a data-toggle="modal" data-target="#docModal" data-keyboard="true"><i class="fa fa-book"></i> <span data-text="Documentation">Documentation</span></a></li>
+                        <li><a data-toggle="modal" data-target="#contactModal" data-keyboard="true"><i class="fa fa-envelope"></i> <span data-text="Contact">Contact</span></a></li>
+                    </ul>
                     <div class="pull-right well well-sm helpMessage">
                         <span data-text="Notice_Mistake">Notice a mistake?</span> 
                         <a data-toggle="modal" data-target="#infoModal" data-text="Help_Improve" data-keyboard="true">Help us improve Apertium!</a>
@@ -245,35 +251,93 @@
             </div>
         </div>
 
-        <div class="modal fade" id="infoModal" tabindex="-1">
-            <div class="modal-dialog">
+        <div class="modal fade" id="aboutModal" tabindex="-1">
+            <div class="modal-dialog modal-lg">
                 <div class="modal-content">
                     <div class="modal-header">
                         <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
-                        <h4 class="modal-title" data-text="About_Title">About the Website</h4>
+                        <h4 class="modal-title" data-text="About_Apertium">About Apertium</h4>
                     </div>
                     <div class="modal-body">
+                        <h4 style="margin-top: 0">What is Apertium?</h4>
+
                         <p>
-                            <span data-text="Maintainer">
-                                This website is maintained by <a href="http://wiki.apertium.org/wiki/Apertium" target="_blank">Apertium</a>.
-                            </span>
-                            <span class="pull-right">
-                                <a href="mailto:apertium-contact@lists.sourceforge.net">
-                                    <i class="glyphicon glyphicon-envelope"></i>
-                                    apertium-contact@lists.sourceforge.net
-                                </a>
-                            </span>
+                            Apertium is a <b>free/open-source machine translation platform</b>, initially aimed at related-language pairs but recently expanded to deal with more divergent language pairs (such as English-Catalan). The platform provides
                         </p>
-                        <p data-text="Contact_Us">
-                            Feel free to contact us if you find a mistake, there's a project you would like to see us work on, or you would like to help out.
+    
+                        <ol>
+                            <li>a language-independent machine translation engine</li>
+                            <li>tools to manage the linguistic data necessary to build a machine translation system for a given language pair and</li>
+                            <li>linguistic data for a growing number of language pairs.</li>
+                        </ol>
+
+                        <p>
+                            Apertium uses a shallow-transfer machine translation engine which processes the input text in stages, as in an assembly line: de-formatting, morphological analysis, part-of-speech disambiguation, shallow structural transfer, lexical transfer, morphological generation, and re-formatting.
                         </p>
+                        <p>
+                            Apertium uses finite-state transducers for all lexical processing operations (morphological analysis and generation, lexical transfer), hidden Markov models for part-of-speech tagging, and multi-stage finite-state based chunking for structural transfer.
+                        </p>
+                        <p>
+                            The initial design was largely based upon that of systems already developed by the Transducens group at the Universitat d'Alacant, such as <b><a href="http://www.internostrum.com/welcome.php" target="_blank">interNOSTRUM</a></b> (Spanish-Catalan, and <b><a href="http://traductor.universia.net/" target="_blank">Traductor Universia</a></b> (Spanish-Portuguese).
+                        </p>
+                        <p>
+                            It is possible to use Apertium to build machine translation systems for a variety of language pairs; to that end, Apertium uses simple XML-based standard formats to encode the linguistic data needed (either by hand or by converting existing data), which are compiled using the provided tools into the high-speed formats used by the engine.
+                        </p>
+
+                        <h4>Who is developing it?</h4>
+
+                        <p>
+                            The Apertium architecture is being developed by the <a href="http://transducens.dlsi.ua.es/" target="_blank">Transducens</a> research group at the <a href="http://www.dlsi.ua.es/" target="_blank">Departament de Llenguatges i Sistemes Informàtics</a> of the <a href="http://www.ua.es/" target="_blank">Universitat d'Alacant</a> in collaboration with <a href="http://www.prompsit.com/" target="_blank">Prompsit Language Engineering</a>.
+                        </p>
+                            
+                        <p>Linguistic data for Apertium are being developed by Transducens and Prompsit</p>
+                        <ul>
+                            <li>Spanish–Catalan</li>
+                            <li>Spanish–Portuguese</li>
+                            <li>Catalan–French</li>
+                            <li>Occitan–Catalan</li>
+                            <li>English–Catalan</li>
+                        </ul>
+                        the <a href="http://webs.uvigo.es/sli/" target="_blank">Seminario de Lingüística Informática</a> of the <a href="http://www.uvigo.es/indice/index.gl.htm" target="_blank">Universidade de Vigo</a>
+                        <ul>
+                            <li>Spanish–Galician</li>
+                        </ul>
+                        and the <a href="http://www.iula.upf.edu/" target="_blank">Institut Universitari de Lingüística Aplicada</a> of the <a href="http://www.upf.edu/" target="_blank">Universitat Pompeu Fabra</a> in Barcelona
+                        <ul>
+                            <li>Occitan–Catalan</li>
+                            <li>English–Catalan</li>
+                            <li>French–Catalan</li>
+                        </ul>
                         
-                        <div class="center-block" style="margin-bottom: 1.5em;">
-                            <a href="http://www.apertium.org/" target="_blank">
-                                <img alt="Apertium" data-src="./assets/img/Apertium.png" class="center-block" style="height: 5em;">
-                            </a>
-                        </div>
+                        <p>
+                            <a href="http://sourceforge.net/project/memberlist.php?group_id=143781" target="_blank">Many other developers</a> are also generating <a href="http://ilazki.thinkgeek.co.uk/apertium/graph.svg" target="_blank">new pairs</a>. You can <a href="http://apertium.svn.sourceforge.net/viewvc/apertium/" target="_blank">browse the <abbr title="Subversion">SVN</abbr> repository</a> to see a snapshot of current development. 
+                        </p>
+                        <p>
+                            Apertium welcomes new developers: if you think you can improve the engine or the tools, or develop linguistic data for us, do not hesitate to <a href="http://xixona.dlsi.ua.es/apertium-www/?id=contact" target="_blank">contact us</a>.
+                        </p>
 
+                        <h4>Funding</h4>
+                        
+                        <p>
+                            Apertium is one of the two open-source machine translation engines whose development started inside project <a href="http://www.opentrad.org/" target="_blank">OpenTrad</a> (2004–2005); Apertium is designed to translate between related languages. The OpenTrad consortium was led by <a href="http://www.eleka.net/" target="_blank">Eleka Ingeniaritza Linguistikoa</a>.
+                        </p>
+                        
+                        <p>
+                            The Opentrad project ("Open Source Machine Translation for the languages of Spain") was funded by the <a href="http://www.min.es/" target="_blank">Ministry of Industry, Tourism and Commerce</a> of Spain through PROFIT grant numbers FIT340101-2004-0003 and FIT340001-2005-0002.
+                        </p>
+                        
+                        <p>
+                            More recently, Apertium has also received funding from the <a href="http://www10.gencat.net/dursi/AppJava/home.jsp?area=3&amp;=2" target="_blank">Secretariat for Telecommunications and Information Society</a> of the Generalitat de Catalunya (the government of the autonomous community of Catalonia in Spain) to develop new language pairs and an improved architecture (Apertium 2.0) to include more difficult pairs such as English–Catalan, and again from the <a href="http://www.min.es/" target="_blank">Ministry of Industry, Tourism and Commerce</a> of Spain through <a href="http://www.mityc.es/profit/" target="_blank">PROFIT</a> grant number FIT350401-2006-05 (EurOpenTrad).
+                        </p>
+                        
+                        <p>
+                            Project Trautorom ("Romanian-Spanish machine translation", package <code>apertium-es-ro</code>) has been funded by: the Universitat d'Alacant through its Department of Computer Languages and Systems and the Office of the Vice-President for Extracurricular Activities (Vicerrectorado de Extensión Universitaria) and by the Romanian <a href="http://www.mae.ro/" target="_blank">Ministry of Foreign Affairs</a>.
+                        </p>
+                        
+                        <p>Parts of Apertium have also been funded by the <a href="http://www.ua.es/" target="_blank">Universitat d'Alacant</a>.</p>
+
+                        <br>
+
                         <div class="row lead">
                             <span class="col-md-6">
                                 <div class="center-block">
@@ -339,6 +403,98 @@
             </div>
         </div>
 
+        <div class="modal fade" id="docModal" tabindex="-1">
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                        <h4 class="modal-title" data-text="Apertium_Documentation">Apertium Documentation</h4>
+                    </div>
+                    <div class="modal-body">
+                        <p>
+                            Documentation can be found on our <a href="http://wiki.apertium.org/">Wiki</a> under the <a href="http://wiki.apertium.org/wiki/Documentation" target="_blank">Documentation</a> sub-page. We have published various conference papers and journal articles, a list of which may be found on the Wiki under <a href="http://wiki.apertium.org/wiki/Publications" target="_blank">Publications.
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <div class="modal fade" id="downloadModal" tabindex="-1">
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                        <h4 class="modal-title" data-text="Apertium_Downloads">Apertium Downloads</h4>
+                    </div>
+                    <div class="modal-body">
+                        <p>
+                            Current versions of the <b>Apertium toolbox</b> as well as of <b>language-pair data</b> are available from <a href="https://sourceforge.net/projects/apertium/" target="_blank">the SourceForge page</a>. Installation instructions for Apertium on all major platforms are provided on the <a href="http://wiki.apertium.org/wiki/Installation" target="_blank">Wiki's Installation page</a>.
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <div class="modal fade" id="contactModal" tabindex="-1">
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                        <h4 class="modal-title" data-text="Contact">Contact Us</h4>
+                    </div>
+                    <div class="modal-body">
+                        <h4>IRC channel</h4>
+                        <p>
+                            The quickest way to contact us is by joining our <a href="http://wiki.apertium.org/wiki/IRC" target="_blank"><abbr title="Internet Relay Chat">IRC</abbr> channel, #apertium</a> at irc.freenode.net, where users and developers of Apertium meet. You don't need an IRC client; use can <a href="http://webchat.freenode.net/?channels=apertium">freenode webchat</a>.
+                        </p>
+
+                        <h4>Mailing list</h4>
+                        <p>
+                            Also, subscribe to the <a href="https://lists.sourceforge.net/lists/listinfo/apertium-stuff" target="_blank"> apertium-stuff mailing list</a>, where you can post longer proposals or issues, as well as follow general Apertium discussions.
+                        </p>
+
+                        <h4>Contact</h4>
+                        <p>
+                            Feel free to contact us via the <a href="mailto:apertium-contact@lists.sourceforge.net" target="_blank">apertium-contact mailing list</a> if you find a mistake, there's a project you would like to see us work on, or you would like to help out.
+                        </p>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <div class="modal fade" id="infoModal" tabindex="-1">
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                        <h4 class="modal-title" data-text="About_Title">About the Website</h4>
+                    </div>
+                    <div class="modal-body">
+                        <p>
+                            <span data-text="Maintainer">
+                                This website is maintained by <a href="http://wiki.apertium.org/wiki/Apertium" target="_blank">Apertium</a>.
+                            </span>
+                            <span class="pull-right">
+                                <a href="mailto:apertium-contact@lists.sourceforge.net">
+                                    <i class="glyphicon glyphicon-envelope"></i>
+                                    apertium-contact@lists.sourceforge.net
+                                </a>
+                            </span>
+                        </p>
+                        <p data-text="Contact_Us">
+                            Feel free to contact us if you find a mistake, there's a project you would like to see us work on, or you would like to help out.
+                        </p>
+                        
+                        <div class="center-block" style="margin-bottom: 1.5em;">
+                            <a href="http://www.apertium.org/" target="_blank">
+                                <img alt="Apertium" data-src="./assets/img/Apertium.png" class="center-block" style="height: 5em;">
+                            </a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+
         <div id="noscript">
             <div class="lead text-warning well well-sm">
                 This site only works with JavaScript enabled, if you cannot <a href="http://www.enable-javascript.com/" target="_blank">enable Javascript</a>, then try the 
